agents/wiregrid_encoder/signal_parser.py

```diff
@@ -0,0 +1,257 @@
+import socket
+import struct
+import select
+import time
+import calendar
+import numpy as np
+from collections import deque
+import txaio
+txaio.use_twisted()
+
+# should be consistent with the software on beaglebone
+COUNTER_INFO_LENGTH = 100
+# header, quad, clock[100], clock_overflow[100], refcount[100], error[100]
+COUNTER_PACKET_SIZE = 4 + 4 * 5 * COUNTER_INFO_LENGTH
+# header, clock, clock_overflow, info[10], synch[10], synch_overflow[10]
+IRIG_PACKET_SIZE = 132
+# header, type
+TIMEOUT_PACKET_SIZE = 8
+
+
+class EncoderParser:
+
+    def __init__(self, beaglebone_port=50007, read_chunk_size=8192):
+
+        self.encoder_queue = deque()
+        self.irig_queue = deque()
+
+        self.is_start = 1
+        self.start_time = [0, 0, 0]
+        self.current_time = 0
+
+        self.sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        self.sock.bind(('', beaglebone_port))
+
+        self.data = ''
+        self.read_chunk_size = read_chunk_size
+
+        self.log = txaio.make_logger()
+
+    def check_once(self):
+
+        enc_oncetime = False
+        irig_oncetime = False
+        checking_start = time.time()
+        sampling_timeout = 30
+
+        while True:
+            ready = select.select([self.sock], [], [], 2)
+            if ready[0]:
+                self.data = self.sock.recv(self.read_chunk_size)
+            if not self.check_data_length(0, 4):
+                self.log.error(
+                    'Header data length error in check_once()')
+            header = self.data[0:4]
+            header = struct.unpack('<I', header)[0]
+
+            if enc_oncetime and irig_oncetime:
+                print('---finished taking each data---')
+                break
+            else:
```

Comment:
You don't really need this else and can get rid of it and un-indent the below section. If the above if statement is entered you'll hit that `break`.

Reply:
Done